nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/processors/ConsumeJMS.java

```diff
@@ -136,9 +155,16 @@
         relationships = Collections.unmodifiableSet(_relationships);
     }
 
+    @OnScheduled
```

Comment:
Hey @mosermw is this something we need to do? We have the Character Set Validator already in place so it should never get this far if it's not valid.

Reply:
Thanks for looking at this @markap14.  The CHARACTER_SET_VALIDATOR essentially says a property value is valid if an EL expression returns a String.  If someone sets it to ${system.charset} and that environment variable is set to "FOO", for example, then when ConsumeJMS receives a TextMessage it will throw an UnsupportedCharsetException at runtime.  I thought I would include this method to give earlier warning.  PublishJMS doesn't need this check, because the charset could be set in a flowfile attribute, and UnsupportedCharsetException would just cause the flowfile to go to 'failure'.  If you still think the method is unnecessary, though, let me know and I can remove it.

Reply:
Very good point, and I like the customValidate approach.  I'll make the change.